crates/subspace-proof-of-time/src/pot.rs

```diff
@@ -0,0 +1,101 @@
+//! Common defines.
+
+use sp_core::{Blake2Hasher, Hasher, H256};
+use std::ops::BitXor;
+
+/// The 128 bit key for the AES encryption.
+#[derive(Clone, Debug)]
+pub struct AesKey(pub [u8; 16]);
+
+impl From<&AesSeed> for AesKey {
+    fn from(seed: &AesSeed) -> AesKey {
+        Self(h256_to_arr(Blake2Hasher::hash(&seed.0)))
+    }
+}
+
+/// Input to AES.
+#[derive(Clone, Debug)]
+pub struct AesSeed(pub [u8; 16]);
+
+impl From<H256> for AesSeed {
+    fn from(hash: H256) -> Self {
+        Self(h256_to_arr(hash))
+    }
+}
+
+/// Output from AES.
+#[derive(Clone, Debug)]
+pub struct AesCipherText(pub [u8; 16]);
+
+/// Config params for PoT.
+#[derive(Debug)]
+pub struct PotConfig {
+    /// Frequency of entropy injection from consensus.
+    pub randomness_update_interval_blocks: u32,
+
+    /// Starting point for entropy injection from consensus.
+    pub injection_depth_blocks: u32,
+
+    /// Number of slots it takes for updated global randomness to
+    /// take effect.
+    pub global_randomness_reveal_lag_slots: u32,
+
+    /// Number of slots it takes for injected randomness to
+    /// take effect.
+    pub pot_injection_lag_slots: u32,
+
+    /// Number of checkpoints per proof.
+    pub num_checkpoints: u32,
+
+    /// Number of EAS iterations per checkpoints.
+    /// Total iterations per proof = num_checkpoints * checkpoint_iterations.
+    pub checkpoint_iterations: u32,
+}
+
+/// Proof of time.
+/// TODO: versioning.
+#[derive(Debug)]
```

Comment:
You will almost certainly want to store it somewhere and more around, `Clone` is a must here.

And this should also live in `subspace-core-primitives`. Think about it this way: someone in the ecosystem might want to write an alternative implementation of the proof of time, what types would they need to do this? Those fundamental types (that you will see in block header, network messages, consensus runtime state, etc.) should live in `subspace-core-primitives`, while type specific to particular implementation and library used should live in this crate.

Reply:
Done